03_building_and_packaging/pip_demo.md

```diff
@@ -0,0 +1,85 @@
+# `pip`: Python Packaging Tool Demo Notes
+
+**Note**: Maybe run everything in a fresh Ubuntu Docker container. Launch such a container using `docker run --rm -it ubuntu:jammy`. The other option is to work in a Python virtual environment. An environment can be created as follows
+
+```bash
+python -m venv env-name
+source env-name/bin/activate
+```
+
+## 0. Getting `pip`
+
+If Python has been installed using a system package manager like `apt`, it will not come with `pip`. `pip`can be installed in several ways
+
+```bash
+python -m ensurepip --upgrade
+```
+
+or
+
+```bash
+apt install pip
+```
+
+- If installed with Python, the Python version is used to determine which version of `pip` is installed. So Python 3.8 will install `pip3.8`.
```

Comment:
pip has an independent versioning

```
$ pip3 --version
pip 22.2.2 from /home/uekermbn/.local/lib/python3.8/site-packages/pip (python 3.8)
```

Reply:
Ah indeed this is confusing! The version I point to is the Python version to which a particular `pip` is compatible with, but of course `pip` itself has independent versioning. I will clarify this in the same point.